### PR TITLE
EUI-4373: 16-digit case reference input validation

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
[EUI-4373](https://tools.hmcts.net/jira/browse/EUI-4373)

### Change description ###
Implementation of validator for the case reference search box, and error page routing on validation failure.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnbenjamin-hmcts — APPROVED** [review]

File: src/app/components/hmcts-global-header/hmcts-global-header.component.html [hunk 1]
```diff
@@ -1,54 +1,55 @@
...
-                  <img class="img-responsive" alt="Gov UK Logo" src="/assets/images/govuk-crest-jcm.png" width="83px">
-                </a>
-              </div>
-              <div class="govuk-header__logo">
-              </div>
-              <a class="hmcts-header__link" [routerLink]="headerTitle?.url" [attr.aria-label]="headerTitle?.name" >{{headerTitle?.name}}</a>
-          </ng-container>
+      <!-- Logo for Judicial Users -->
+      <ng-container *ngSwitchCase="'judicial'">
+        <div style="position: absolute;" *ngIf="showItems">
```
Comment (paulhowes-HMCTS) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.html:9]: Can we put this inline style into the scss file?

File: src/app/components/hmcts-global-header/hmcts-global-header.component.html [hunk 2]
```diff
@@ -57,8 +58,8 @@
     <div class="hmcts-primary-navigation__nav">
       <nav class="hmcts-primary-navigation transparent-background" aria-label="Primary navigation">
         <ul class="hmcts-primary-navigation__list">
-          <li class="hmcts-primary-navigation__item" *ngFor="let item of (leftItems | async)" >
-            <a class="hmcts-primary-navigation__link" (click)="onEmitSubMenu(item)" [attr.aria-current]="item.active ? true: null" [routerLink]="item.href">{{item.text}}</a>
+          <li class="hmcts-primary-navigation__item" *ngFor="let item of (leftItems | async)">
+            <a class="hmcts-primary-navigation__link" (click)="onEmitSubMenu(item)" [attr.aria-current]="item.active ? true : null" [routerLink]="item.href">{{item.text}}</a>
```
Comment (paulhowes-HMCTS) [inline src/app/components/hmcts-global-header/hmcts-global-header.component.html:62]: Recommend setting attr.aria-current to false as opposed to null (unless when set to null it does not render the attribute)

**Review by paulhowes-HMCTS — APPROVED** [review]
A couple of potential tweaks / queries but nothing to prevent approval.

**Review by Mo-Lala84 — APPROVED** [review]
approved

File: src/app/components/hmcts-global-header/hmcts-global-header.component.html (hunk 1 above)
Comment (Daniel-Lam) [reply src/app/components/hmcts-global-header/hmcts-global-header.component.html]: Hi @paulhowes-HMCTS - I've moved the inline style into the accompanying SCSS file.

File: src/app/components/hmcts-global-header/hmcts-global-header.component.html (hunk 2 above)
Comment (Daniel-Lam) [reply src/app/components/hmcts-global-header/hmcts-global-header.component.html]: Hi @paulhowes-HMCTS - I've changed the setting to `false` for both `attr.aria-current` instances, when `!item.active`.

Comment (paulhowes-HMCTS) [reply src/app/components/hmcts-global-header/hmcts-global-header.component.html]: Great @Daniel-Lam thanks.

**Pull request merged**